doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+:math:`k - 1`, as was the case for the non-padded convolution. This is
+sometimes referred to as ``full`` padding, because every element of the
+kernel is superimposed on every element of the input feature map. Here is an
+example for :math:`i = 5`, :math:`k = 3` and (therefore) :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/full_padding_no_strides.gif
+
+Strides
+=======
+
+Up until now, we only considered convolutions where the kernel was translated by
+increments of :math:`s = 1`. Let's now examine what happens when :math:`s > 1`.
+To keep things simple, let's ignore padding for the moment, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p = 0`.
+
+Here is what the strided convolution looks like for :math:`i = 5`, :math:`k = 3`
+and :math:`s = 2`:
+
+.. figure:: conv_arithmetic_figures/no_padding_strides.gif
+
+Once again, we are counting the number of positions where the kernel can be
+placed.  Let's consider the x-axis: the left side of the kernel starts at the
+left side of the input, but this time we slide the kernel **by hops of**
+:math:`s` until its right side touches the right side of the input. The size of
+the output will be equal to the number of steps we need to make, plus one,
+because we need to account for the initial position of the kernel. The same
+applies for the y-axis.
+
+From this, we conclude that the output size :math:`o` of the strided convolution
+is
+
+.. math::
+
+    o = \left\lfloor \frac{i - k}{s} \right\rfloor + 1.
+
+Combining padding and strides
+-----------------------------
+
+The most general case, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p`,
+
+is obtained by applying both the rule we derived for strides and the one we
+derived for the padding. In other words,
+
+.. math::
+
+    o = \left\lfloor \frac{i + 2p - k}{s} \right\rfloor + 1.
+
+Here is what it looks like for :math:`i = 5`, :math:`k = 3`, :math:`s = 2` and
+:math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides.gif
+
+Sometimes numbers are mean
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may have noticed that the floor function in the general case formula will
+produce the same output shape for multiple input shapes. More specifically, if
+:math:`i + 2p - k` is a multiple of :math:`s`, then any input size
+:math:`j = i + a, \quad a \in \{0,\ldots,s - 1\}` will produce the same output
+size. Note that this ambiguity applies only for :math:`s > 1`.
+
+As an example, here is what a convolution looks like for :math:`i = 6`,
+:math:`k = 3`, :math:`s = 2` and :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides_odd.gif
+
+Despite having different input sizes, this convolution and the previous
+convolution share the same output size.
+
+This may seem like a curiosity, but keep that in mind, as it will become
+important when we examine *transposed convolutions*.
```

Comment:
This is as much as we need here (very nice!), so regarding the previous comment I would definitely move that paragraph to the transposed convolution section.